artifacts/source_type.py

```diff
@@ -271,3 +271,20 @@ def __init__(self, query=None, base_object=None):
     super(WMIQuerySourceType, self).__init__()
     self.query = query
     self.base_object = base_object
+
+
+SOURCE_TYPES = {
```

Comment:
Yes - see https://codereview.appspot.com/297590043/patch/40001/50012

I dont understand what you mean - why would adding custom types defeat the purpose of verification? The new types must be verified - thats kind of the point of adding them to the registry?